Add io.github.d0ksan8.zen-app-manager

d0ksan8 opened the pull request:
<!-- ⚠️⚠️  Submission pull request MUST be made against the `new-pr` **base branch** ⚠️⚠️  -->

<!-- 💡 Please go to the preview tab to view the markdown below 💡 -->

### Please confirm your submission meets all the criteria

This application requires --filesystem=xdg-config/autostart to create and remove its own autostart .desktop file when the user enables or disables the autostart option.
The application only manages its own file and does not interact with other system files.

<!-- 💡 Please replace each `[ ]` with `[X]` when the step is complete 💡 -->

<!-- 💡 Please tick and write 'N/A' with a reason if a checklist item below is not applicable 💡 -->

- [x] Please describe the application briefly.  Zen App Manager is a minimal and modern tool designed to help you manage your startup applications on Linux.
- [x] Please attach a video showcasing the application on Linux using the Flatpak. 

https://github.com/user-attachments/assets/f90eae0f-06f9-48ad-bf07-5e5f618c815c


- [x] The Flatpak ID follows all the rules listed in the [Application ID requirements][appid].
- [x] I have read and followed all the [Submission requirements][reqs] and the [Submission guide][reqs2] and I agree to them.
- [x] I am an author to the project.

<!-- 💡 Please mention below the GitHub usernames of any additional maintainers needed (if any) 💡 -->

<!-- ⚠️⚠️  Please DO NOT modify anything below this line ⚠️⚠️  -->

[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[reqs2]: https://docs.flathub.org/docs/for-app-authors/submission

Commit:
Add com.zen.zen-app-manager

Commit:
Translate comments to English

Commit:
Add cargo config for offline builds

Commit:
Update commit hash for Registry support and UI improvements

Commit:
Update commit hash to latest version

Commit:
Add README with acknowledgements

Commit:
Update metainfo with latest features

Commit:
Sync README with main repo

Commit:
Sync clean README

Commit:
Rename App ID to io.github.d0ksan8.zen-app-manager and fix permissions

Commit:
Update to v0.3.0 (Systemd support)

Commit:
Update commit hash to include UI fixes

Commit:
Add second screenshot to metainfo

Commit:
Sync README with System Services feature

Commit:
Update to latest commit (Performance fix)

hfiguiere commented on zen-app-manager.desktop:
```diff
@@ -0,0 +1,8 @@
+Type=Application
+Name=Zen App Manager
+Comment=Manage your startup applications
+Exec=/home/zen/devzone/ZEN/zen-app-manager/src-tauri/target/release/zen-app-manager
```
and how do you think this work outside of your machine?

hfiguiere commented on zen-app-manager.desktop:
```diff
@@ -0,0 +1,8 @@
+Name=Zen App Manager
+Comment=Manage your startup applications
+Exec=/home/zen/devzone/ZEN/zen-app-manager/src-tauri/target/release/zen-app-manager
+Icon=/home/zen/devzone/ZEN/zen-app-manager/src-tauri/icons/128x128.png
```
same.

d0ksan8 replied:
done

hfiguiere commented on io.github.d0ksan8.zen-app-manager.yml:
```diff
@@ -0,0 +1,37 @@
+      - install -D -m644 src-tauri/icons/128x128.png /app/share/icons/hicolor/128x128/apps/io.github.d0ksan8.zen-app-manager.png
+      # Adjust desktop file for Flatpak
+      - sed -i 's|^Exec=.*|Exec=zen-app-manager|' zen-app-manager.desktop
+      - sed -i 's|^Icon=.*|Icon=io.github.d0ksan8.zen-app-manager|' zen-app-manager.desktop
```
Never use sed to change desktop files. It has no failure mode. Use the command `desktop-file-edit`.

d0ksan8 replied:
done

Commit:
Address review comments: Remove flathub.json, use upstream metainfo, update runtime to 49

Commit:
Address review: Use upstream desktop file, remove sed, update commit hash

d0ksan8 commented:
Thanks for the detailed review! I've addressed all the points:

- Removed [flathub.json](cci:7://file:///home/zen/devzone/flathub_submission/flathub.json:0:0-0:0).
- Moved both [.desktop](cci:7://file:///home/zen/devzone/flathub_submission/zen-app-manager.desktop:0:0-0:0) and [.metainfo.xml](cci:7://file:///home/zen/devzone/flathub_submission/com.zen.zen-app-manager.metainfo.xml:0:0-0:0) files upstream to the main repository.
- Removed `sed` usage in the manifest; now installing the upstream desktop file directly.
- Updated `runtime-version` to 49.
- Cleaned up the git source definition (removed tag).

Ready for another look

d0ksan8 commented:
bot,build

d0ksan8 commented on zen-app-manager.desktop:
done

Commit:
Update commit hash: Fix UI freezing and symlink issues

Commit:
Update commit hash: Fix command execution bug

Commit:
Fix: Remove unnecessary :create suffix from filesystem permissions

d0ksan8 commented:
@hfiguiere  The linter failures for "unnecessary-xdg-*-autostart-rw-access" are false positives for this app. Since this is a "Startup App Manager", it explicitly needs direct RW access to autostart directories to function. It cannot use portals for this core feature.

d0ksan8 commented:
@hfiguiere Okay. Since you state they are not false positives, could you please point me to the correct XDG Portal or API that allows listing, toggling, and deleting *existing* autostart entries? as a "startup manager", the app needs to see and modify what's already there. f there's a standard way to do this without filesystem access, I'm more than happy to implement it.

d0ksan8 commented:
@hfiguiere I've looked into other similar applications on Flathub to see how they handle this.

For example, "Startup Configuration" (best.ellie.StartupConfiguration) also requests direct filesystem access in its manifest: --filesystem=xdg-config/autostart:create --filesystem=xdg-data/autostart:create

And in its source code (Rust), it writes directly to these directories without using Portals: 


`// From best.ellie.StartupConfiguration source:
#[cfg(feature = "flatpak")]
match std::fs::write(directory_to_target.join(file_name), ...)`

Since my application (Zen App Manager) has the exact same purpose and functionality, I believe it requires the same permissions exception to function correctly.

bbhtt commented on io.github.d0ksan8.zen-app-manager.yml:
```diff
@@ -0,0 +1,34 @@
+  - --device=dri
+  # Permissions required to manage startup applications
+  - --filesystem=xdg-config/autostart
+  - --filesystem=xdg-data/autostart
```
config/autostart is the standard location where is this coming from?

bbhtt commented on io.github.d0ksan8.zen-app-manager.yml:
```diff
@@ -0,0 +1,34 @@
+  - --socket=wayland
+  - --device=dri
+  # Permissions required to manage startup applications
+  - --filesystem=xdg-config/autostart
```
needs an [exception](https://docs.flathub.org/docs/for-app-authors/linter#exceptions)

d0ksan8 replied:
thanks <3

d0ksan8 commented:
@bbhtt Exactly. This application is a "Startup Manager". Its core functionality is to list, toggle, and delete startup entries found in the standard directories (~/.config/autostart and ~/.local/share/autostart).

Since the current XDG Portals do not provide an API to manage/delete existing autostart entries, direct filesystem access to these locations is required for the app to function.

Could you please guide me on how to properly request/add this exception? Should I open a separate PR to flatpak-builder-lint or can it be handled here?

Commit:
Remove unnecessary xdg-data/autostart permission

Commit:
Update source commit to v0.3.2

d0ksan8 commented:
The video below shows the latest and most up-to-date version of the application.


https://github.com/user-attachments/assets/97b05606-58ce-4231-834b-cd9d2675a86b

Commit:
build: update commit hash and generated sources for winreg fix

d0ksan8 commented:
bot, build

Commit:
fix: restore dconf permissions to prevent GTK init failure

hfiguiere commented on io.github.d0ksan8.zen-app-manager.yml:
```diff
@@ -0,0 +1,39 @@
+app-id: io.github.d0ksan8.zen-app-manager
+runtime: org.gnome.Platform
+runtime-version: '47'
```
this runtime is EOL. This will not be accepted.

Commit:
build: update runtime version to 49 as requested

d0ksan8 commented:
bot, build

Commit:
build: switch to runtime version 48 for stability

Commit:
build: revert to runtime version 49

Commit:
build: remove dconf permissions to pass linter

Commit:
fix: use memory backend for gsettings to prevent crash without dconf

d0ksan8 commented:
@bbhtt  Which operating system are you using? And could you please describe your problem in detail?

Commit:
fix: force GDK_BACKEND=x11 to prevent crashes

Commit:
fix: replace X11 enforcement with WEBKIT_DISABLE_DMABUF_RENDERER=1 for proper Wayland support

Commit:
Fix: update commit hash and add gsk_renderer=g1 for VM compatibility

Commit:
Fix: downgrade runtime directly to 47 for stability

Commit:
Fix: revert to runtime 49 and use GSK_RENDERER=cairo for max compat

d0ksan8 commented:
bot, build

Commit:
Fix: update commit hash for metainfo release order fix

Commit:
update: bump upstream commit hash to fix titlebar and path issues

Commit:
fix(manifest): update source commit and add talk-name permission

Commit:
chore: update manifest with latest source commit
